lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+from flask_restful import Resource, abort
+from flask import request, make_response, jsonify
+from pbench.server.database.models.metadata import Metadata, MetadataKeys
+from pbench.server.api.auth import Auth
+
+
+class UserMetadata(Resource):
+    """
+    Abstracted pbench API for handling user metadata by using graphql schema
+    """
+
+    def __init__(self, config, logger, auth):
+        self.server_config = config
+        self.logger = logger
+        self.auth = auth
+
+    @Auth.token_auth.login_required(optional=True)
+    def post(self, key):
+        """
+        Post request for creating metadata instance for a user.
+
+        The url requires a metadata session key such as /metadata/<key>
+        the only keys acceptable to create the metadata sessions are favourite/saved
+
+        This requires a JSON data with required user metadata fields
+        {
+            "value": "blog text" <Client defined text, can be a string of json>,
+        }
+        Example: {"value": '{"config": "config", "description": "description"}'}
+
+        Authorization header can be included as
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+        If the authorization header is not present, the created metadata session becomes public by default
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
+                    "data" {
+                        "id": "metadata_id",
+                        "value": "client text blob",
+                        "created": "datetime string",
+                        "updated": "datetime string",
+                        "key": favorite/saved
+                        }
+                }
+        """
+        metadata_key = key.upper()
+        if metadata_key not in [key.name for key in MetadataKeys]:
+            self.logger.warning(
+                "Invalid Metadata key provided during metadata creation. Key: {}",
+                metadata_key,
+            )
+            abort(
+                400,
+                message="Invalid metadata key. Key need to be either Favorite/Saved",
+            )
+
+        post_data = request.get_json()
+        if not post_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(400, message="Invalid json object in request")
+
+        current_user = self.auth.token_auth.current_user()
+        if current_user:
+            current_user_id = current_user.id
+        else:
+            current_user_id = None
+
+        value = post_data.get("value")
+        if not value:
+            self.logger.warning(
+                "value not provided during metadata creation. user_id: {}",
+                current_user_id,
+            )
+            abort(400, message="Value field missing")
+
+        try:
+            # Create a new metadata session
```

Comment:
"Session" still bothers me. It implies a sequence, a continuity; we're just creating a record in a database associating their value with a key. Are you really tied to this terminology?

Reply:
Fixed, I changed sessions to just metadata objects so it can take any form